### ISSUE TITLE
Deleting a file downloads it first

### ISSUE THREAD

**Issue by mappu** [kurin/blazer#77]
Using v0.5.3.

If I delete a file in a straightforward way:

```golang
obj := myBucket.Object(name)
return obj.Delete(ctx)
```

This involves [a call to ensure()](https://github.com/kurin/blazer/blob/master/b2/b2.go#L573);

```golang
	if err := o.ensure(ctx); err != nil {
		return err
	}
	return o.f.deleteFileVersion(ctx)
```

which calls `downloadFileByName`, downloading the entire file and discarding the body.

This is slow and bandwidth-wasteful - is there any other way to implement ensure()?

**Comment by kurin** [comment]
Hm, I seem to recall this got fixed a while back.  It may be that I just never cut a release after fixing it, so try it with head.  I'll check more closely when I get a minute.

Also, I don't know your situation, but this library uses v1 of the API, and apparently there's now a v2, and also they claim there's S3-API compatibility, which means tools written against S3 should work on B2, so consider if it's worth migrating in that direction as this library doesn't see a ton of maintenance and essentially zero new development.

**Comment by mappu** [comment]
> Hm, I seem to recall this got fixed a while back. It may be that I just never cut a release after fixing it, so try it with head. I'll check more closely when I get a minute.

You're right, the fix for #71 changed ensure() to make HEAD requests.

I'll track head, and thanks for the advice about v2 api.
